### PR TITLE
rate limiter per event name

### PR DESCRIPTION
**Pull request by sagikSF**
add LIMIT_FOR_PERIOD value for tests

### CODE REVIEW

File: timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/cron/CronsRunner.java [hunk 1]
```diff
@@ -76,10 +79,11 @@ private void runEventsPersistentFetchCron(String eventsPersistentFetchCronExp, P
 		scheduler.scheduleJob(job, trigger);
 	}
 
-	private void runBulkPersistentFetchCron(String bulkPersistentFetchCronExp, ElasticsearchClient es, PersistenceHandler persistenceHandler) throws SchedulerException {
+	private void runBulkPersistentFetchCron(String bulkPersistentFetchCronExp, ElasticsearchClient es, PersistenceHandler persistenceHandler, LoadingCache<String, RateLimiter> rateLimiterMap) throws SchedulerException {
```
Comment (edengal) [inline timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/cron/CronsRunner.java:82]: Rate limit is not needed in this cron, can be removed.

File: timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/cron/CronsRunner.java [hunk 2]
```diff
@@ -24,17 +26,17 @@
...
 	public void runCrons(String bulkPersistentFetchCronExp, String eventsPersistentFetchCronExp, PersistenceHandler persistenceHandler, ElasticsearchClient es, String deletionCronExp, BlockingQueue<Event> buffer,
-						 BlockingQueue<Event> overFlowedEvents, String mergingCronExp, RedisService redisService) {
+						 BlockingQueue<Event> overFlowedEvents, String mergingCronExp, RedisService redisService, LoadingCache<String, RateLimiter> rateLimiterMap) {
 		final StdSchedulerFactory sf = new StdSchedulerFactory();
 		try {
 			 scheduler = sf.getScheduler();
 			if (persistenceHandler != null) {
 				if (!Strings.isEmpty(bulkPersistentFetchCronExp)) {
-					runBulkPersistentFetchCron(bulkPersistentFetchCronExp, es, persistenceHandler);
+					runBulkPersistentFetchCron(bulkPersistentFetchCronExp, es, persistenceHandler, rateLimiterMap);
```
Comment (edengal) [inline timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/cron/CronsRunner.java:35]: no need

File: timbermill-java/timbermill-server/src/test/java/com/datorama/oss/timbermill/TimberLogAdvancedTest.java [hunk 3]
```diff
@@ -99,6 +99,26 @@ public void testOngoingTask() {
         assertEquals(string3, ongoingTask.getString().get(string3));
     }
 
+    public void testTaskReachedRateLimit() {
```
Comment (edengal) [inline timbermill-java/timbermill-server/src/test/java/com/datorama/oss/timbermill/TimberLogAdvancedTest.java:102]: As this is not using TimberLoggerAdvanced class, let's move it to TimberLogTest

File: timbermill-java/timbermill-server/src/test/java/com/datorama/oss/timbermill/TimberLogAdvancedTest.java [hunk 4]
```diff
@@ -99,6 +99,26 @@ public void testOngoingTask() {
...
+        String ongoingTaskName = EVENT + "_rate_limit";
+        String limitStr = System.getenv("LIMIT_FOR_PERIOD");
```
Comment (edengal) [inline timbermill-java/timbermill-server/src/test/java/com/datorama/oss/timbermill/TimberLogAdvancedTest.java:104]: Don't see the point in that, just set a value

**Review by edengal — CHANGES_REQUESTED** [review]

File: timbermill-java/timbermill-server/src/test/java/com/datorama/oss/timbermill/TimberLogAdvancedTest.java (hunk 3 above)
Comment (edengal) [inline timbermill-java/timbermill-server/src/test/java/com/datorama/oss/timbermill/TimberLogAdvancedTest.java:102]: Also add a test the after enough time has passed the timbermill will be able to process the limited tasks again

File: timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillService.java [hunk 5]
```diff
@@ -92,7 +98,10 @@ public TimbermillService(@Value("${INDEX_BULK_SIZE:200000}") Integer indexBulkSi
 							 @Value("${REDIS_POOL_MAX_IDLE:50}") int redisPoolMaxIdle,
 							 @Value("${REDIS_POOL_MAX_TOTAL:50}") int redisPoolMaxTotal,
 							 @Value("${REDIS_MAX_TRIED:3}") int redisMaxTries,
-							 @Value("${FETCH_BY_IDS_PARTITIONS:10000}") int fetchByIdsPartitions){
+							 @Value("${FETCH_BY_IDS_PARTITIONS:10000}") int fetchByIdsPartitions,
+                             @Value("${LIMIT_FOR_PERIOD:10000000}") int limitForPeriod,
```
Comment (edengal) [inline timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillService.java:102]: Too high let's put it at 30000 per minute

Comment (sagikSF) [reply timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillService.java:102]: done

File: timbermill-java/timbermill-server/src/test/java/com/datorama/oss/timbermill/TimberLogAdvancedTest.java (hunk 4 above)
Comment (sagikSF) [reply timbermill-java/timbermill-server/src/test/java/com/datorama/oss/timbermill/TimberLogAdvancedTest.java]: we spoke about this

**Review by edengal — APPROVED** [review]

**Pull request merged**